src/EnergyPlus/VentilatedSlab.cc

```diff
@@ -2765,129 +2762,151 @@ namespace VentilatedSlab {
-                                                 state.dataVentilatedSlab->VentSlab(Item).HCoilSchedValue,
-                                                 state.dataVentilatedSlab->VentSlab(Item).HCoil_Index);
-                    } else if (SELECT_CASE_var1 == Heating_GasCoilType) {
-                        CheckHeatingCoilSchedule(state,
-                                                 "Coil:Heating:Fuel",
-                                                 state.dataVentilatedSlab->VentSlab(Item).HCoilName,
-                                                 state.dataVentilatedSlab->VentSlab(Item).HCoilSchedValue,
-                                                 state.dataVentilatedSlab->VentSlab(Item).HCoil_Index);
-                    } else {
-                    }
-                }
-
-            } else if (SELECT_CASE_var == CoolingOption) {
-
-                {
-                    auto const SELECT_CASE_var1(state.dataVentilatedSlab->VentSlab(Item).CCoilType);
-
-                    if (SELECT_CASE_var1 == Cooling_CoilWaterCooling) {
-                        CheckWaterCoilSchedule(state,
-                                               "Coil:Cooling:Water",
-                                               state.dataVentilatedSlab->VentSlab(Item).CCoilName,
-                                               state.dataVentilatedSlab->VentSlab(Item).CCoilSchedValue,
-                                               state.dataVentilatedSlab->VentSlab(Item).CCoil_Index);
-                    } else if (SELECT_CASE_var1 == Cooling_CoilDetailedCooling) {
-                        CheckWaterCoilSchedule(state,
-                                               "Coil:Cooling:Water:DetailedGeometry",
-                                               state.dataVentilatedSlab->VentSlab(Item).CCoilName,
-                                               state.dataVentilatedSlab->VentSlab(Item).CCoilSchedValue,
-                                               state.dataVentilatedSlab->VentSlab(Item).CCoil_Index);
-                    } else if (SELECT_CASE_var1 == Cooling_CoilHXAssisted) {
-                        CheckHXAssistedCoolingCoilSchedule(state,
-                                                           "CoilSystem:Cooling:Water:HeatExchangerAssisted",
-                                                           state.dataVentilatedSlab->VentSlab(Item).CCoilName,
-                                                           state.dataVentilatedSlab->VentSlab(Item).CCoilSchedValue,
-                                                           state.dataVentilatedSlab->VentSlab(Item).CCoil_Index);
-                    } else {
-                    }
-                }
+        switch (ventSlab.coilOption) {
+        case CoilType::Both: {
+
+            switch (ventSlab.hCoilType) {
+            case HeatingCoilType::Water: {
+                CheckWaterCoilSchedule(state,
+                                       "Coil:Heating:Water",
```

Comment:
I was wondering if this switch/case could be collapsed because the only difference is this argument, and it turns out that ... this argument is not even used! 🤣

Reply:
But the functions are different (`CheckWaterCoilSchedule` vs `CheckSteamCoilSchedule` vs `CheckHeatingCoilSchedule`) in each case, right?

Reply:
Details... 😭